source/testing/testing-components.md

```diff
@@ -25,74 +25,122 @@ export default Component.extend({
 Pretty Color: {{name}}
 ```
 
-The `moduleForComponent` helper will find the component by name (`pretty-color`)
-and its template (if available).  Make sure to set `integration: true` to enable
-integration test capability.
+The `module` from QUnit will scope your tests into groups of tests which can be
+configured and run independently. Make sure to call the `setupRenderingTest` function together with the `hooks` parameter
+first in your new module. This will do all the setup necessary for testing your component for you, including setting up a way to access your component later on in the test and it will clean up after you once your tests in this module are finished.
 
 ```tests/integration/components/pretty-color-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
 
-moduleForComponent('pretty-color', 'Integration | Component | pretty color', {
-  integration: true
 });
 ```
+Inside of your `module` and after setting up the test, we can now start to create our first test case. Here we can use the `QUnit.test` helper and we can
+give it a descriptive name:
 
-Each test following the `moduleForComponent` call has access to the `render()`
-function, which lets us create a new instance of the component by declaring
-the component in template syntax, as we would in our application.
+```tests/integration/components/pretty-color-test.js
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
 
-We can test that changing the component's `name` property updates the
-component's `style` attribute and is reflected in the  rendered HTML:
+  test('should change colors', async function(assert) {
+
+
+  });
+});
+```
+Also note how the callback function passed to the test helper is marked with the keyword `async`. The [ECMAScript 2017 feature async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) allows us to write asynchronous code in an easy-to-read, seemingly synchronous manner. We can better see what this means, once we start writing out our first test case:
 
 ```tests/integration/components/pretty-color-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 
-moduleForComponent('pretty-color', 'Integration | Component | pretty color', {
-  integration: true
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('should change colors', async function(assert) {
+    assert.expect(1);
```

Comment:
we could probably remove this line to make the test simpler.

my rule of thumb for using `assert.expect()` is to use it (only) when assertions are happening in callbacks. async/await is something like callbacks essentially but since it ensures that we always return a Promise I don't think it's needed in this case.